### ISSUE TITLE
Base58.decode should return Maybe or Result

### ISSUE THREAD

**Issue by cmditch** [nathanjohnson320/base58#1]
Erroneous input is currently valid.

```
> Base58.decode ""
Zer : BigInt.BigInt

> Base58.decode ")(*&^%$#@!"
Zer : BigInt.BigInt

> Base58.decode "a)(*&^%$#@!a"
Pos (Magnitude [2747169,5200546,8245592]) : BigInt.BigInt

> Base58.decode "aa"
Pos (Magnitude [1947]) : BigInt.BigInt
```

Much like `BigInt.fromString` returns `Maybe BigInt`, or  [truqu's](http://package.elm-lang.org/packages/truqu/elm-base64/latest) `Base64.decode` returning a `Result`.

Let me know if your keen on this change, and I can start work on a PR.

Thanks!
Coury

**Comment by najohnson** [comment]
@cmditch I'd be cool with the `Maybe` route if you're up to put in the PR

**Comment by nathanjohnson320** [comment]
Published as 2.0.0 here: http://package.elm-lang.org/packages/nathanjohnson320/base58/2.0.0
